### PR TITLE
Fix F821 flake error: undefined variable

### PR DESCRIPTION
**Pull request by santisoler**
#### Summary

Remove the `F821` warning from the ignored flake8 warning lists. Solve a few undefined variables across the repository. Remove the unused `DerivProjfieldsTest` test function.

#### PR Checklist
* [ ] If this is a work in progress PR, set as a Draft PR
* [ ] Linted my code according to the [style guides](https://docs.simpeg.xyz/content/getting_started/contributing/code-style.html).
* [ ] Added [tests](https://docs.simpeg.xyz/content/getting_started/practices.html#testing) to verify changes to the code.
* [ ] Added necessary documentation to any new functions/classes following the
      expect [style](https://docs.simpeg.xyz/content/getting_started/practices.html#documentation).
* [ ] Marked as ready for review (if this is was a draft PR), and converted
      to a Pull Request
* [ ] Tagged ``@simpeg/simpeg-developers`` when ready for review.

### CODE REVIEW

**Comment by santisoler** [comment]
I think allowing flake8 to warn us about undefined variables is an important quality check we should have. This means not ignoring the F821 warnings that flake8 throws. I tackled the ones that looked more obvious to me, but there are a few places where things aren't that obvious. @lheagy, maybe you could give me a hand on solving those.

You can find a list of undefined variables detected by flake8 by running `make check` on this branch, or in the results of Azure pipeline: https://dev.azure.com/simpeg/simpeg/_build/results?buildId=4382&view=logs&j=bb50e881-bb1d-58eb-be78-4cb911813d1f&t=db5183b1-eaf4-58a7-a474-f325080a5c59

I'm copying it just in case:

```
7.0.0 (flake8-bugbear: 23.12.2, flake8-builtins: 2.2.0, flake8-docstrings:
1.7.0, flake8-mutable: 1.2.0, flake8-rst-docstrings: 0.3.0, mccabe: 0.7.0,
pycodestyle: 2.11.1, pyflakes: 3.2.0) CPython 3.10.12 on Linux
flake8  simpeg examples tutorials tests
examples/20-published/plot_heagyetal2017_casing.py:953:38: F821 undefined name 'mesh'
simpeg/electromagnetics/analytics/FDEMDipolarfields.py:202:45: F821 undefined name 'E_galvanic_from_ElectricDipoleWholeSpaced'
simpeg/electromagnetics/analytics/FDEMDipolarfields.py:232:9: F821 undefined name 'E_inductive_from_ElectricDipoleWholeSpaced'
simpeg/electromagnetics/analytics/FDEMDipolarfields.py:321:10: F821 undefined name 'mu'
simpeg/electromagnetics/analytics/FDEMDipolarfields.py:322:10: F821 undefined name 'mu'
simpeg/electromagnetics/analytics/FDEMDipolarfields.py:323:10: F821 undefined name 'mu'
simpeg/electromagnetics/analytics/FDEMcasing.py:125:11: F821 undefined name 'omega'
simpeg/electromagnetics/natural_source/fields.py:684:38: F821 undefined name 'v'
simpeg/electromagnetics/natural_source/survey.py:156:19: F821 undefined name 'Planewave_xy_1DhomotD'
simpeg/electromagnetics/natural_source/utils/plot_utils.py:694:72: F821 undefined name 'error'
simpeg/electromagnetics/natural_source/utils/source_utils.py:46:52: F821 undefined name 'ez_py'
simpeg/electromagnetics/natural_source/utils/source_utils.py:116:52: F821 undefined name 'ez_py'
tests/em/nsem/inversion/test_Problem3D_Derivs.py:116:13: F821 undefined name 'rx'
tests/em/nsem/inversion/test_Problem3D_Derivs.py:117:23: F821 undefined name 'rx'
tests/em/tdem/test_TDEM_crosscheck.py:45:29: F821 undefined name 'prb'
tests/em/tdem/test_TDEM_crosscheck.py:46:28: F821 undefined name 'prb'
```

File: simpeg/electromagnetics/natural_source/survey.py [hunk 1]
```diff
@@ -152,8 +152,8 @@ def fromRecArray(cls, recArray, srcType="primary"):
         """
         if srcType == "primary":
             src = PlanewaveXYPrimary
-        elif srcType == "total":
-            src = Planewave_xy_1DhomotD
+        # elif srcType == "total":
+        #     src = Planewave_xy_1DhomotD
```
Comment (lheagy) [inline simpeg/electromagnetics/natural_source/survey.py:156]: we don't currently have a total field formulation, so I think we can leave this commented out for now

File: tests/em/nsem/inversion/test_Problem3D_Derivs.py [hunk 2]
```diff
@@ -91,35 +91,6 @@ def fun(x):
     return tests.check_derivative(fun, m, num=3, plotIt=False, eps=FLR)
 
 
-def DerivProjfieldsTest(inputSetup, comp="All", freq=False):
```
Comment (lheagy) [inline tests/em/nsem/inversion/test_Problem3D_Derivs.py:94]: this wasn't run, and with all of the development work that @dccowan is doing in #1454, a lot more thorough testing will be added to the NSEM module, so I think we are okay removing this

**Comment by santisoler** [comment]
Thanks @lheagy for the changes you pushed. I just fixed how the `omega` was being used since you define it as a numeric variable and in the return it was being called ([a9c6b36](https://github.com/simpeg/simpeg/pull/1487/commits/a9c6b3641b5ddf3a799a5f283ecbfd71d2eb6bc0)), and improved the docstrings for the method in the NSEM Data class to reflect the changes you pushed.

If CI pass, I'll merge this.

**Pull request merged**